sycl/doc/UsersManual.md

```diff
@@ -1,63 +1,120 @@
 
-    Pass <arg> to the SYCL based target backend.
+**`-f[no-]sycl-dead-args-optimization`**
 
-**`-Xsycl-target-frontend=<triple> <arg>`**
+    Enables (or disables) LLVM IR dead argument elimination pass to remove
+    unused arguments for the kernel functions before translation to SPIR-V.
+    Disabled by default.
 
-    Pass <arg> to the SYCL based target frontend identified by <triple>.
+**`-f[no-]sycl-id-queries-fit-in-int`**
 
-**`-Xsycl-target-frontend <arg>`**
+    Assume/Do not assume that SYCL ID queries fit within MAX_INT. It assumes
+    that these values fit within MAX_INT:
+    * id class get() member function and operator[]
+    * item class get_id() member function and operator[]
+    * nd_item class get_global_id()/get_global_linear_id() member functions
+    Enabled by default.
 
-    Pass <arg> to the SYCL based target frontend.
+## Target toolchain options
 
-**`-Xsycl-target-linker=<triple> <arg>`**
+**`-Xsycl-target-backend=<T> "options"`**
+**`-Xs "options"`**
 
-    Pass <arg> to the SYCL based target linker identified by <triple>.
+    Pass "options" to the backend of target device compiler, specified by
+    triple T. The backend of device compiler generates target machine code from
+    intermediate representation. This option can be used to tune code
+    generation for a specific target. The "options" are used during offline
+    compilation and are also saved in a fat binary for online compilation.
+    -Xs is a shortcut to pass "options" to all backends specified via the
+    '-fsycl-targets' option (or default one).
 
-**`-Xsycl-target-linker <arg>`**
+**`-Xsycl-target-frontend=<T> "options"`**
 
-    Pass <arg> to the SYCL based target linker.
+    Pass "options" to the frontend of target device compiler, specified by
+    triple T. This option can be used to control of intermediate representation
+    generation during offline or online compilation.
 
-### Link options
+**`-Xsycl-target-linker=<T> "options"`**
+
+    Pass "options" to the device code linker, when linking multiple device
+    object modules. T is specific target device triple.
+
+## Link options
 
 **`-fsycl-link`**
 
-    Generate partially linked device object to be used with the host link.
+    Link device object modules and wrap those into a host-compatible object
+    module that can be linked later by any standard host linker into the final
+    fat binary.
 
-**`-fsycl-link-targets=<T1,...,Tn>`**
+**`-fsycl-link-targets=<T1,...,Tn>`** [DEPRECATED]
 
     Specify comma-separated list of triples SYCL offloading targets to produce
     linked device images. Used in a link step to link device code for given
     targets and output multiple linked device code images, whose names consist
     of the common prefix taken from the -o option and the triple string.
     Does not produce fat binary and must be used together with -fsycl.
 
-**`-fsycl-add-targets=<T1:file1...Tn:filen>`**
+**`-fsycl-add-targets=<T1:file1...Tn:filen>`** [DEPRECATED]
 
```

Comment:
For `-fsycl-[link|add]-targets` are there any current users of these?  It would seem the desired approach is to use our regular AOT compilation model.

Reply:
Confirmed with FPGA folks, should be OK to deprecate.